src/SkipStages.cpp

```diff
@@ -78,37 +78,48 @@ class PredicateFinder : public IRVisitor {
         op->body.accept(this);
         if (should_pop) {
             varying.pop(op->name);
-            //internal_assert(!expr_uses_var(predicate, op->name));
+            // internal_assert(!expr_uses_var(predicate, op->name));
```

Comment:
We should either remove this line or add a comment about why we aren't doing this check.